src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/DateFiltersTests.cs

```diff
@@ -15,12 +15,45 @@ public class DateFiltersTests
     {
         public static IEnumerable<object[]> GetValidDataForAddHyphensDate()
         {
-            yield return new object[] { null, null };
-            yield return new object[] { string.Empty, string.Empty };
-            yield return new object[] { @"2001", @"2001" };
-            yield return new object[] { @"200101", @"2001-01" };
-            yield return new object[] { @"19241010", @"1924-10-10" };
-            yield return new object[] { @"19850101000000", @"1985-01-01" };
+            yield return new object[] { null, "local", null };
+            yield return new object[] { string.Empty, "local", string.Empty };
+            yield return new object[] { @"2001", "preserve", @"2001" };
+            yield return new object[] { @"200101", "preserve", @"2001-01" };
+            yield return new object[] { @"19241010", "local", @"1924-10-10" };
+            yield return new object[] { @"19850101000000", "local", @"1985-01-01" };
+        }
+
+        // We assume the local timezone is +08:00.
+        public static IEnumerable<object[]> GetValidLocalDateTimeForAddHyphensDateWithUTCTimeZoneHandling()
+        {
+            yield return new object[] { @"200101", "utc", @"2000-12" };
+            yield return new object[] { @"20010102", "utc", @"2001-01-01" };
+            yield return new object[] { @"19850101000000", "utc", @"1984-12-31" };
+        }
+
+        public static IEnumerable<object[]> GetValidDataForAddSeconds()
+        {
+            yield return new object[] { null, 60, "local", null };
+            yield return new object[] { string.Empty, 60, "local", string.Empty };
+
+            yield return new object[] { @"1970-01-01T00:01:00+10:00", -60, "utc", @"1969-12-31T14:00:00Z" };
+
+            yield return new object[] { @"1970-01-01T00:01:00Z",  60, "preserve", @"1970-01-01T00:02:00Z" };
+            yield return new object[] { @"1970-01-01T00:01:00+06:00", 60, "preserve", @"1970-01-01T00:02:00+06:00" };
+            yield return new object[] { @"2001-01", 60, "preserve", @"2001-01-01T00:01:00" };
+            yield return new object[] { @"1970-01-01T00:01:00+14:00", 60, "utc", @"1969-12-31T10:02:00Z" };
+
+            // Skip this test in pipeline, as the local time zone is different
+            // yield return new object[] { @"1924-10-10", 60000, "utc", @"1924-10-10T08:40:00Z" };
+            // yield return new object[] { @"1970-01-01T00:01:00+06:00", 60, "local", @"1970-01-01T02:02:00+08:00" };
+            // yield return new object[] { @"1924-10-10", 60000, "local", @"1924-10-10T16:40:00" };
```

Comment:
Are these test cases still work after the update? (We can try these cases locally.)

Reply:
Yes, they work locally.